agent.go

```diff
@@ -38,8 +38,9 @@ type Agent struct {
 	retries     RetryManager
 	vbRouter    VbucketRouter
 
-	httpCfgWatcher *ConfigWatcherHttp
-	memdCfgWatcher *ConfigWatcherMemd
+	httpCfgWatcher   *ConfigWatcherHttp
+	memdCfgWatcher   *ConfigWatcherMemd
+	cfgWatcherCancel func()
```

Comment:
I wonder if we should just have an 'agentBackgroundContext' that covers all background work that an Agent is directly performing.  I guess that doesn't work well for waiting for them to all finish though.